### ISSUE TITLE
[Q]Why it has create lots of runs based on one project 

### ISSUE THREAD

**Issue by Yiir-1** [wandb/wandb#6354]
as u can see, I have created a lot of number of runs based on one step of running a project，I wonder why？

![image](https://github.com/wandb/wandb/assets/81577270/03a6e64b-d0e3-429b-a740-c8ff0794052a)

the runs number has up to 10!

![image](https://github.com/wandb/wandb/assets/81577270/b99b6bb9-1d32-4bf4-a6eb-f56ffa5a8bb9)

I would appreciate it if u can reply me soon, hope u have a good day (●'◡'●)。

**Comment by Yiir-1** [comment]
my code is as follows:
`
def train_worker(rank, addr, port):
    
    # Distributed Setup
    os.environ['MASTER_ADDR'] = addr
    os.environ['MASTER_PORT'] = port
    dist.init_process_group("nccl", rank=rank, world_size=distributed_num_gpus)
    
    # Training DataLoader
    dataset_train = ZipDataset([
 
    ])
    dataset_train_len_per_gpu_worker = int(len(dataset_train) / distributed_num_gpus)
    dataset_train = Subset(dataset_train, range(rank * dataset_train_len_per_gpu_worker, (rank + 1) * dataset_train_len_per_gpu_worker))
    dataloader_train = DataLoader(dataset_train,
                                  shuffle=True,
                                  pin_memory=True,
                                  drop_last=True,
                                  batch_size=args.batch_size // distributed_num_gpus,
                                  num_workers=args.num_workers // distributed_num_gpus)
    
    # Validation DataLoader
    if rank == 0:
        dataset_valid = ZipDataset([
           
        ])
        dataset_valid = SampleDataset(dataset_valid, 50)
        dataloader_valid = DataLoader(dataset_valid,
                                      pin_memory=True,
                                      drop_last=True,
                                      batch_size=args.batch_size // distributed_num_gpus,
                                      num_workers=args.num_workers // distributed_num_gpus)
    
    # Model
    model = MattingRefine().to(rank)
    model = nn.SyncBatchNorm.convert_sync_batchnorm(model)
    model_distributed = nn.parallel.DistributedDataParallel(model, device_ids=[rank],find_unused_parameters=True)
    
    if args.model_last_checkpoint is not None:
        load_matched_state_dict(model, torch.load(args.model_last_checkpoint))

    optimizer = Adam()
    scaler = GradScaler()
    
    # Logging and checkpoints
    if rank == 0:
        if not os.path.exists(f'checkpoint/{args.model_name}'):
            os.makedirs(f'checkpoint/{args.model_name}')

    
    # Run loop
    for epoch in range(args.epoch_start, args.epoch_end):
        for i, ((true_pha, true_fgr), true_bgr) in enumerate(tqdm(dataloader_train)):
            
            # Composite foreground onto source
            true_src = true_fgr * true_pha + true_src * (1 - true_pha)
            with autocast():
                pred_pha, pred_fgr, pred_pha_sm, pred_fgr_sm, pred_err_sm, _ = model_distributed(true_src, true_bgr)
                loss = compute_loss()

            scaler.scale(loss).backward()
            scaler.step(optimizer)
            scaler.update()
            optimizer.zero_grad()

            if rank == 0:
                if (i + 1) % args.log_train_loss_interval == 0:
                    wandb.log({
                        "Train loss": loss / i
                    })

           
                if (i + 1) % args.log_valid_interval == 0:
                    valid(model, dataloader_valid, step)

                if (step + 1) % args.checkpoint_interval == 0:
                    torch.save(model.state_dict(), f'checkpoint/{args.model_name}/epoch-{epoch}-iter-{step}.pth')
                    
        if rank == 0:
            torch.save(model.state_dict(), f'checkpoint/{args.model_name}/epoch-{epoch}.pth')
            
    # Clean up
    dist.destroy_process_group()
            


# --------------- Start ---------------


if __name__ == '__main__':
    addr = 'localhost'
    port = str(random.choice(range(12300, 12400))) # pick a random port.
    mp.spawn(train_worker,
             nprocs=distributed_num_gpus,
             args=(addr, port),
             join=True)`

**Comment by Yiir-1** [comment]
maybe I know the question is, if I used the unparallel code, the question was solved. But I wonder how to use parallel code with wandb?

**Comment by luisbergua** [comment]
Hi @Yiir-1, thanks for sharing this! So you're trying to create only one run, is that correct? Here you have two sections ([1](https://docs.wandb.ai/guides/track/tracking-faq#how-can-i-use-wandb-with-multiprocessing-eg-distributed-training) and [2](https://docs.wandb.ai/guides/track/log/distributed-training)) from our docs containing information about this that I think would be helpful for you. Please let me know if you have any other questions!

**Comment by sydholl** [comment]
WandB Internal User commented: 
Yiir-1 commented: 
maybe I know the question is, if I used the unparallel code, the question was solved. But I wonder how to use parallel code with wandb?

**Comment by sydholl** [comment]
WandB Internal User commented: 
luisbergua commented: 
Hi @Yiir-1, thanks for sharing this! So you're trying to create only one run, is that correct? Here you have two sections ([1](https://docs.wandb.ai/guides/track/tracking-faq#how-can-i-use-wandb-with-multiprocessing-eg-distributed-training) and [2](https://docs.wandb.ai/guides/track/log/distributed-training)) from our docs containing information about this that I think would be helpful for you. Please let me know if you have any other questions!

**Comment by sydholl** [comment]
Luis Bergua commented: 
Hi there, I wanted to follow up on this request. Please let us know if we can be of further assistance or if your issue has been resolved.

**Comment by luisbergua** [comment]
Hi @Yiir-1 , since we have not heard back from you we are going to close this request. If you would like to re-open the conversation, please let us know!
